src/main/java/gregtech/api/metatileentity/TieredMetaTileEntity.java

```diff
@@ -68,6 +86,47 @@ public void renderMetaTileEntity(CCRenderState renderState, Matrix4 translation,
         getBaseRenderer().render(renderState, translation, colouredPipeline);
     }
 
+    @Override
+    public void update() {
+        super.update();
+        if (ConfigHolder.machines.doEnvironmentExplosion) {
+            checkEnvironmentExplosion();
+        }
+    }
+
+    public void checkEnvironmentExplosion() {
+        World world = getWorld();
+        if (!world.isRemote && !getEnvironmentResistance() && energyContainer.getEnergyStored() != 0) {
+            if (world.rand.nextInt(1000) == 0) {
+                for (EnumFacing side : EnumFacing.VALUES) {
+                    Block block = getWorld().getBlockState(getPos().offset(side)).getBlock();
+                    if (block == Blocks.FIRE || block == Blocks.WATER || block == Blocks.FLOWING_WATER || block == Blocks.LAVA || block == Blocks.FLOWING_LAVA) {
+                        doExplosion(tier, true);
+                        return;
+                    }
+                }
+            }
+            if (world.rand.nextInt(1000) == 0) {
+                if (world.isRainingAt(getPos()) || world.isRainingAt(getPos().east()) || world.isRainingAt(getPos().west()) || world.isRainingAt(getPos().north()) || world.isRainingAt(getPos().south())) {
+                    if (world.isThundering() && world.rand.nextInt(3) == 0) {
+                        doExplosion(tier, true);
```

Comment:
Hm, I think that the terrain damage here should still respect the config option `doExplosion`. All the config option does is determine if the terrain is damaged when the machine explodes, it will explode on either config option.